Lower version bounds on `crucible-llvm-0.7.1` and `crux-llvm-0.9.1` are too permissive

`crucible-llvm-0.7.1` (as uploaded to Hackage) requires `what4-0.7` or later in order to compile, but its lower version bounds in its `.cabal` file do not reflect this:

https://github.com/GaloisInc/crucible/blob/aa46418b427dd2ace7f447074f29a01945f30227/crucible-llvm/crucible-llvm.cabal#L43

This can lead to a failing build plan, as seen below:

<details>

```
$ cabal repl -b crucible-llvm --constraint="crucible-llvm==0.7.1" --constraint="what4<1.7"
Resolving dependencies...
Build profile: -w ghc-9.4.8 -O1
In order, the following will be built (use -v for more details):
 - crucible-llvm-0.7.1 (lib) (requires build)
 - fake-package-0 (interactive) (lib) (first run)
Starting     crucible-llvm-0.7.1 (lib)
Building     crucible-llvm-0.7.1 (lib)

Failed to build crucible-llvm-0.7.1.
Build log (
/Users/rscott/.cache/cabal/logs/ghc-9.4.8/crcbl-llvm-0.7.1-23d2aeef.log ):
Configuring library for crucible-llvm-0.7.1...
Preprocessing library for crucible-llvm-0.7.1...
Building library for crucible-llvm-0.7.1...
[ 1 of 58] Compiling Lang.Crucible.LLVM.Arch.Util ( src/Lang/Crucible/LLVM/Arch/Util.hs, dist/build/Lang/Crucible/LLVM/Arch/Util.o, dist/build/Lang/Crucible/LLVM/Arch/Util.dyn_o )
[ 2 of 58] Compiling Lang.Crucible.LLVM.Arch.X86 ( src/Lang/Crucible/LLVM/Arch/X86.hs, dist/build/Lang/Crucible/LLVM/Arch/X86.o, dist/build/Lang/Crucible/LLVM/Arch/X86.dyn_o )
ld: warning: -single_module is obsolete
[ 3 of 58] Compiling Lang.Crucible.LLVM.Bytes ( src/Lang/Crucible/LLVM/Bytes.hs, dist/build/Lang/Crucible/LLVM/Bytes.o, dist/build/Lang/Crucible/LLVM/Bytes.dyn_o )
[ 4 of 58] Compiling Lang.Crucible.LLVM.DataLayout ( src/Lang/Crucible/LLVM/DataLayout.hs, dist/build/Lang/Crucible/LLVM/DataLayout.o, dist/build/Lang/Crucible/LLVM/DataLayout.dyn_o )
[ 5 of 58] Compiling Lang.Crucible.LLVM.Errors.Standards ( src/Lang/Crucible/LLVM/Errors/Standards.hs, dist/build/Lang/Crucible/LLVM/Errors/Standards.o, dist/build/Lang/Crucible/LLVM/Errors/Standards.dyn_o )
[ 6 of 58] Compiling Lang.Crucible.LLVM.Extension.Arch ( src/Lang/Crucible/LLVM/Extension/Arch.hs, dist/build/Lang/Crucible/LLVM/Extension/Arch.o, dist/build/Lang/Crucible/LLVM/Extension/Arch.dyn_o )
[ 7 of 58] Compiling Lang.Crucible.LLVM.Intrinsics.Match ( src/Lang/Crucible/LLVM/Intrinsics/Match.hs, dist/build/Lang/Crucible/LLVM/Intrinsics/Match.o, dist/build/Lang/Crucible/LLVM/Intrinsics/Match.dyn_o )
[ 8 of 58] Compiling Lang.Crucible.LLVM.Intrinsics.Options ( src/Lang/Crucible/LLVM/Intrinsics/Options.hs, dist/build/Lang/Crucible/LLVM/Intrinsics/Options.o, dist/build/Lang/Crucible/LLVM/Intrinsics/Options.dyn_o )
[ 9 of 58] Compiling Lang.Crucible.LLVM.MalformedLLVMModule ( src/Lang/Crucible/LLVM/MalformedLLVMModule.hs, dist/build/Lang/Crucible/LLVM/MalformedLLVMModule.o, dist/build/Lang/Crucible/LLVM/MalformedLLVMModule.dyn_o )
[10 of 58] Compiling Lang.Crucible.LLVM.MemModel.Options ( src/Lang/Crucible/LLVM/MemModel/Options.hs, dist/build/Lang/Crucible/LLVM/MemModel/Options.o, dist/build/Lang/Crucible/LLVM/MemModel/Options.dyn_o )
[11 of 58] Compiling Lang.Crucible.LLVM.MemModel.Type ( src/Lang/Crucible/LLVM/MemModel/Type.hs, dist/build/Lang/Crucible/LLVM/MemModel/Type.o, dist/build/Lang/Crucible/LLVM/MemModel/Type.dyn_o )
[12 of 58] Compiling Lang.Crucible.LLVM.MemModel.Common ( src/Lang/Crucible/LLVM/MemModel/Common.hs, dist/build/Lang/Crucible/LLVM/MemModel/Common.o, dist/build/Lang/Crucible/LLVM/MemModel/Common.dyn_o )
[13 of 58] Compiling Lang.Crucible.LLVM.PrettyPrint ( src/Lang/Crucible/LLVM/PrettyPrint.hs, dist/build/Lang/Crucible/LLVM/PrettyPrint.o, dist/build/Lang/Crucible/LLVM/PrettyPrint.dyn_o )
[14 of 58] Compiling Lang.Crucible.LLVM.MemType ( src/Lang/Crucible/LLVM/MemType.hs, dist/build/Lang/Crucible/LLVM/MemType.o, dist/build/Lang/Crucible/LLVM/MemType.dyn_o )
[15 of 58] Compiling Lang.Crucible.LLVM.Printf ( src/Lang/Crucible/LLVM/Printf.hs, dist/build/Lang/Crucible/LLVM/Printf.o, dist/build/Lang/Crucible/LLVM/Printf.dyn_o )
[16 of 58] Compiling Lang.Crucible.LLVM.Translation.BlockInfo ( src/Lang/Crucible/LLVM/Translation/BlockInfo.hs, dist/build/Lang/Crucible/LLVM/Translation/BlockInfo.o, dist/build/Lang/Crucible/LLVM/Translation/BlockInfo.dyn_o )
[17 of 58] Compiling Lang.Crucible.LLVM.Translation.Options ( src/Lang/Crucible/LLVM/Translation/Options.hs, dist/build/Lang/Crucible/LLVM/Translation/Options.o, dist/build/Lang/Crucible/LLVM/Translation/Options.dyn_o )
[18 of 58] Compiling Lang.Crucible.LLVM.TypeContext ( src/Lang/Crucible/LLVM/TypeContext.hs, dist/build/Lang/Crucible/LLVM/TypeContext.o, dist/build/Lang/Crucible/LLVM/TypeContext.dyn_o )
[19 of 58] Compiling Lang.Crucible.LLVM.Types ( src/Lang/Crucible/LLVM/Types.hs, dist/build/Lang/Crucible/LLVM/Types.o, dist/build/Lang/Crucible/LLVM/Types.dyn_o )
[20 of 58] Compiling Lang.Crucible.LLVM.MemModel.Pointer ( src/Lang/Crucible/LLVM/MemModel/Pointer.hs, dist/build/Lang/Crucible/LLVM/MemModel/Pointer.o, dist/build/Lang/Crucible/LLVM/MemModel/Pointer.dyn_o )
[21 of 58] Compiling Lang.Crucible.LLVM.Translation.Types ( src/Lang/Crucible/LLVM/Translation/Types.hs, dist/build/Lang/Crucible/LLVM/Translation/Types.o, dist/build/Lang/Crucible/LLVM/Translation/Types.dyn_o )
[22 of 58] Compiling Lang.Crucible.LLVM.Translation.Constant ( src/Lang/Crucible/LLVM/Translation/Constant.hs, dist/build/Lang/Crucible/LLVM/Translation/Constant.o, dist/build/Lang/Crucible/LLVM/Translation/Constant.dyn_o )
[23 of 58] Compiling Lang.Crucible.LLVM.Translation.Aliases ( src/Lang/Crucible/LLVM/Translation/Aliases.hs, dist/build/Lang/Crucible/LLVM/Translation/Aliases.o, dist/build/Lang/Crucible/LLVM/Translation/Aliases.dyn_o )
[24 of 58] Compiling Lang.Crucible.LLVM.MemModel.Value ( src/Lang/Crucible/LLVM/MemModel/Value.hs, dist/build/Lang/Crucible/LLVM/MemModel/Value.o, dist/build/Lang/Crucible/LLVM/MemModel/Value.dyn_o )
[25 of 58] Compiling Lang.Crucible.LLVM.MemModel.MemLog ( src/Lang/Crucible/LLVM/MemModel/MemLog.hs, dist/build/Lang/Crucible/LLVM/MemModel/MemLog.o, dist/build/Lang/Crucible/LLVM/MemModel/MemLog.dyn_o )
[26 of 58] Compiling Lang.Crucible.LLVM.MemModel.CallStack.Internal ( src/Lang/Crucible/LLVM/MemModel/CallStack/Internal.hs, dist/build/Lang/Crucible/LLVM/MemModel/CallStack/Internal.o, dist/build/Lang/Crucible/LLVM/MemModel/CallStack/Internal.dyn_o )
[27 of 58] Compiling Lang.Crucible.LLVM.MemModel.CallStack ( src/Lang/Crucible/LLVM/MemModel/CallStack.hs, dist/build/Lang/Crucible/LLVM/MemModel/CallStack.o, dist/build/Lang/Crucible/LLVM/MemModel/CallStack.dyn_o )
[28 of 58] Compiling Lang.Crucible.LLVM.Errors.Poison ( src/Lang/Crucible/LLVM/Errors/Poison.hs, dist/build/Lang/Crucible/LLVM/Errors/Poison.o, dist/build/Lang/Crucible/LLVM/Errors/Poison.dyn_o )
[29 of 58] Compiling Lang.Crucible.LLVM.Errors.MemoryError ( src/Lang/Crucible/LLVM/Errors/MemoryError.hs, dist/build/Lang/Crucible/LLVM/Errors/MemoryError.o, dist/build/Lang/Crucible/LLVM/Errors/MemoryError.dyn_o )
[30 of 58] Compiling Lang.Crucible.LLVM.Errors.UndefinedBehavior ( src/Lang/Crucible/LLVM/Errors/UndefinedBehavior.hs, dist/build/Lang/Crucible/LLVM/Errors/UndefinedBehavior.o, dist/build/Lang/Crucible/LLVM/Errors/UndefinedBehavior.dyn_o )
[31 of 58] Compiling Lang.Crucible.LLVM.Extension.Syntax ( src/Lang/Crucible/LLVM/Extension/Syntax.hs, dist/build/Lang/Crucible/LLVM/Extension/Syntax.o, dist/build/Lang/Crucible/LLVM/Extension/Syntax.dyn_o )
[32 of 58] Compiling Lang.Crucible.LLVM.Extension ( src/Lang/Crucible/LLVM/Extension.hs, dist/build/Lang/Crucible/LLVM/Extension.o, dist/build/Lang/Crucible/LLVM/Extension.dyn_o )
[33 of 58] Compiling Lang.Crucible.LLVM.Errors ( src/Lang/Crucible/LLVM/Errors.hs, dist/build/Lang/Crucible/LLVM/Errors.o, dist/build/Lang/Crucible/LLVM/Errors.dyn_o )
[34 of 58] Compiling Lang.Crucible.LLVM.MemModel.Partial ( src/Lang/Crucible/LLVM/MemModel/Partial.hs, dist/build/Lang/Crucible/LLVM/MemModel/Partial.o, dist/build/Lang/Crucible/LLVM/MemModel/Partial.dyn_o )

src/Lang/Crucible/LLVM/MemModel/Partial.hs:175:48: error:
    Not in scope: data constructor ‘Conjuncts’
    |
175 |       (asApp -> Just (ConjPred (viewConjMap -> Conjuncts tms))) -> go (Fold.toList tms)
    |                                                ^^^^^^^^^

src/Lang/Crucible/LLVM/MemModel/Partial.hs:226:48: error:
    Not in scope: data constructor ‘Conjuncts’
    |
226 |       (asApp -> Just (ConjPred (viewConjMap -> Conjuncts tms))) -> go (Fold.toList tms) NoExplanation
    |                                                ^^^^^^^^^
[36 of 58] Compiling Lang.Crucible.LLVM.Utils ( src/Lang/Crucible/LLVM/Utils.hs, dist/build/Lang/Crucible/LLVM/Utils.o, dist/build/Lang/Crucible/LLVM/Utils.dyn_o )
Error: [Cabal-7125]
Failed to build crucible-llvm-0.7.1 (which is required by fake-package-0). See the build log above for details.
```
</details>

There exists a similar issue for `crux-llvm-0.9.1`. It requires `llvm-pretty-bc-parser-0.5` or later to build, but its `.cabal` file does not reflect this:

https://github.com/GaloisInc/crucible/blob/aa46418b427dd2ace7f447074f29a01945f30227/crux-llvm/crux-llvm.cabal#L100

This can lead to a failing build plan, as seen below:

<details>

```
$ cabal repl -b crux-llvm --constraint="crux-llvm==0.9.1" --constraint="what4==1.7" --constraint="llvm-pretty-bc-parser<0.5"
Resolving dependencies...
Build profile: -w ghc-9.4.8 -O1
In order, the following will be built (use -v for more details):
 - crux-llvm-0.9.1 (lib) (requires build)
 - fake-package-0 (interactive) (lib) (first run)
Starting     crux-llvm-0.9.1 (lib)
Building     crux-llvm-0.9.1 (lib)

Failed to build crux-llvm-0.9.1.
Build log (
/Users/rscott/.cache/cabal/logs/ghc-9.4.8/crx-llvm-0.9.1-f693f6b0.log ):
Configuring library for crux-llvm-0.9.1...
Preprocessing library for crux-llvm-0.9.1...
Building library for crux-llvm-0.9.1...
[1 of 7] Compiling Crux.LLVM.Log    ( src/Crux/LLVM/Log.hs, dist/build/Crux/LLVM/Log.o, dist/build/Crux/LLVM/Log.dyn_o )
[2 of 7] Compiling Crux.LLVM.Overrides ( src/Crux/LLVM/Overrides.hs, dist/build/Crux/LLVM/Overrides.o, dist/build/Crux/LLVM/Overrides.dyn_o )
[3 of 7] Compiling Paths_crux_llvm  ( dist/build/autogen/Paths_crux_llvm.hs, dist/build/Paths_crux_llvm.o, dist/build/Paths_crux_llvm.dyn_o )
[4 of 7] Compiling Crux.LLVM.Config ( src/Crux/LLVM/Config.hs, dist/build/Crux/LLVM/Config.o, dist/build/Crux/LLVM/Config.dyn_o )
[5 of 7] Compiling Crux.LLVM.Simulate ( src/Crux/LLVM/Simulate.hs, dist/build/Crux/LLVM/Simulate.o, dist/build/Crux/LLVM/Simulate.dyn_o )

src/Crux/LLVM/Simulate.hs:35:27: error:
    Module ‘Data.LLVM.BitCode’ does not export ‘ParseWarning(..)’
   |
35 | import Data.LLVM.BitCode (ParseWarning(..), parseBitCodeFromFileWithWarnings,
   |                           ^^^^^^^^^^^^^^^^

src/Crux/LLVM/Simulate.hs:35:45: error:
    Module
    ‘Data.LLVM.BitCode’
    does not export
    ‘parseBitCodeFromFileWithWarnings’
   |
35 | import Data.LLVM.BitCode (ParseWarning(..), parseBitCodeFromFileWithWarnings,
   |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Crux/LLVM/Simulate.hs:36:27: error:
    Module ‘Data.LLVM.BitCode’ does not export ‘ppParseWarnings’
   |
36 |                           ppParseWarnings)
   |                           ^^^^^^^^^^^^^^^
[6 of 7] Compiling Crux.LLVM.Compile ( src/Crux/LLVM/Compile.hs, dist/build/Crux/LLVM/Compile.o, dist/build/Crux/LLVM/Compile.dyn_o )
Error: [Cabal-7125]
Failed to build crux-llvm-0.9.1 (which is required by fake-package-0). See the build log above for details.
```
</details>

--- a/crucible-llvm/crucible-llvm.cabal
+++ b/crucible-llvm/crucible-llvm.cabal
@@ -40,7 +40,7 @@ library
     containers >= 0.5.8.0,
     crucible >= 0.5,
     crucible-symio,
-    what4 >= 0.4.1,
+    what4 >= 0.5,
     extra,
     lens,
     itanium-abi >= 0.1.1.1 && < 0.2,

--- a/crux-llvm/crux-llvm.cabal
+++ b/crux-llvm/crux-llvm.cabal
@@ -97,7 +97,7 @@ library
     config-schema >= 1.2.2.0,
     logict,
     llvm-pretty,
-    llvm-pretty-bc-parser,
+    llvm-pretty-bc-parser >= 0.5,
     mtl,
     parameterized-utils,
     prettyprinter >= 1.7.0

